### ISSUE TITLE
Regex in pattern property is incorrectly parsed

### ISSUE THREAD

**Issue by nicexe** [stoplightio/spectral#1115]
**Describe the bug**
Regex patterns aren't correctly parsed and as a result, valid examples for properties get flagged as non-matching.

**To Reproduce**

1. Here is a schema snippet of an OpenAPI 3 specification:
```yaml
date-of-birth:
  type: string
  format: date
  example: '1991-04-12'
  pattern: '/^(19|20)[0-9]{2}-(0[1-9]|1[0-2])-(0[0-9]|[1-2][0-9]|3[0-1])$/'
```
2. Stoplight Studio flags this as an issue saying:
```
`example` property pattern should match pattern `/^(19|20)[0-9]{2}-(0[1-9]|1[0-2])-(0[0-9]|[1-2][0-9]|3[0-1])$/`
```

**Expected behavior**
The pattern follows the OpenAPI 3 spec which should be in this form ([as per spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schema-object)): [ECMA 262 regular expression](https://www.ecma-international.org/ecma-262/5.1/#sec-7.8.5) and therefore it shouldn't be flagged as an issue.

**Environment (remove any that are not applicable):**
 - Library version: 5.0.0
 - OS: Windows 10
 - Browser: Stoplight Studio 1.11.0

**Comment by nulltoken** [comment]
@nicexe From what I've tested (not through Studio, but through Spectral itself), it will work if you drop the forward slashes that wrap the regex.

```
date-of-birth:
  type: string
  format: date
  example: '1991-04-12'
  pattern: '^(19|20)[0-9]{2}-(0[1-9]|1[0-2])-(0[0-9]|[1-2][0-9]|3[0-1])$'
```

FWIW, the OAS3 schema itself doesn't enclose the regex within slashes

![image](https://user-images.githubusercontent.com/92363/80069105-16328700-8541-11ea-9c22-994b2d038071.png)

**Comment by nicexe** [comment]
the OAS3 spec follows the [ECMA 262 regular expression](https://www.ecma-international.org/ecma-262/5.1/#sec-7.8.5) definition for regex. The `RegularExpressionLiteral` follows this format: `/ RegularExpressionBody / RegularExpressionFlags`
Other tools/parsers such as the [OpenAPI Generator](https://openapi-generator.tech/) not only respect this but also throw an error if the pattern is missing the leading and trailing slashes.

**Comment by nulltoken** [comment]
@nicexe Interesting discussion!

Few thoughts:
- Indeed the specs states _"pattern (This string SHOULD be a valid regular expression, according to the ECMA 262 regular expression dialect)"_. However, it doesn't clearly express if "a valid regular expression" refers to a `RegularExpressionLiteral ` or a `RegularExpressionBody`
- Per the spec, OAS3 `pattern` belongs to a list which title is `The following properties are taken directly from the JSON Schema definition and follow the same specifications:`
 - JSON Schema [doc on regular expression](https://json-schema.org/understanding-json-schema/reference/regular_expressions.html) states *"The regular expression syntax used is from JavaScript (ECMA 262, specifically). However, that complete syntax is not widely supported, therefore it is recommended that you stick to the subset of that syntax described below."*. The provided usage example, through a `pattern` property, isn't enclosed in slashes.
 - This [document](https://github.com/OAI/OpenAPI-Specification/blob/f0b48876f02b32db3ad91894ea5a6842a7976a59/proposals/003_Clarify-Nullable.md#interpretation-of-nullable-false) from the OAS repository shows an example of a schema, exposing a pattern, without any enclosing slashes

**Comment by nulltoken** [comment]
@nicexe Should we keep this issue opened?

**Comment by nicexe** [comment]
I think the pattern, just like in the example in OAS repo that you linked, doesn't require the enclosing slashes.
